assets/javascripts/discourse/components/discourse-reactions-actions-summary.gjs

```diff
@@ -0,0 +1,34 @@
+import Component from "@glimmer/component";
+import { hash } from "@ember/helper";
+import { getOwner } from "@ember/owner";
+import MountWidget from "discourse/components/mount-widget";
+
+export default class ReactionsActionSummary extends Component {
+  static extraControls = true;
+
+  static shouldRender(args) {
+    const site = getOwner(this).lookup("service:site");
```

Comment:
I would expect this to fail? In static methods, `this` refers to the Class itself (not the instance), so it won't have an owner 🤔

Reply:
btw, I did want to set the owner on `context` arg, so that this would work:

```js
static shouldRender(args, context) {
  getOwner(context)
```

I think I have this core change stashed somewhere 😅 

we could make `this` work too/instead but using `context` is less surprising, I think? (though we use `getOwner(this)` almost anywhere so I see why having the same "incantation" here could be nice)

Reply:
Yup agreed - adding the owner on context and/or args makes sense. I'd prefer that to doing anything unusual with `this.`. (I don't think we should be setting owners on Klass objects - they persist beyond the lifetime of the ApplicationInstance)

BTW, worth noting that this particular case, it's not a plugin outlet. We're just using a similar shouldRender pattern for the new post-menu. But still, we should aim for the same ownership approach for this and for plugin outlets 💯

Reply:
> BTW, worth noting that this particular case, it's not a plugin outlet.

Ah, missed that fact 😃 

> I don't think we should be setting owners on Klass objects - they persist beyond the lifetime of the ApplicationInstance

I wouldn't be setting owner on a class itself but doing something like:

```js
setOwner(context, owner);
shouldRender.call(context, args, context);

// vs `shouldRender(args, context);`
// which btw, means `this` currently is `window` I think?
```

Reply:
> `// which btw, means `this` currently is `window` I think?`

In a static method, `this.` refers to the class itself. So you can do stuff like

```
class Blah {
  static shouldRender(){
    return this.someMethod();
  }
  static someMethod(){
    return true;
  }
}
```
so if we use `.call`/`.bind`/`.apply` to change the `this.` value, it'll break this fundamental functionality of static methods.

Reply:
I meant it in the plugin outlet context. This **is** the way we call shouldRender:

https://github.com/discourse/discourse/blob/main/app/assets/javascripts/discourse/app/lib/plugin-connectors.js#L229-L230

Reply:
Ahhh.... Then I think we should fix is so it works as a normal static method 😅

Reply:
The API is now receiving the owner as an explicit parameter and the variables match the (args, context, owner) signature for consistency